### PR TITLE
D8CORE-1664

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- Changed composer to use @dev versions of su-sws packages, added tag_commits_to_master github action, updated readme.
- This PR establishes the `8.x-1.x` branch as our development branch, and changes the release process.
- The new process is to cut a release branch from the latest commit on `8.x-1.x` branch.  Then, update the composer.json file to require the latest tagged versions of the SU-SWS packages, eliminating the `@dev` dependencies.  Then, open a PR to merge the release branch to `master`.  Upon a successful merge to `master`, a new tag will be created to bump the semver here by 1 patch increment.  If you wish to bump it by a minor or major version, include the text `#minor` or `#major` in the PR for the release branch.

# Needed By (Date)
- for 1.1.0 release

# Urgency
- Medium.

# Steps to Test

1. Use this branch in your stack repo.  Run a `composer update --prefer-source`.  Observe that it correctly pulls in the dev versions of our SU-SWS packages.
2. Once merged, open a PR to merge to `master`.  Note that a new semver patch tag is correctly applied.

# Affected Projects or Products
- it indirectly affects acsf-cardinalsites, and because it's pulling in dev versions, it also indirectly affects the SU-SWS dependencies called for in the composer.json

# Associated Issues and/or People
- D8CORE-1664, D8CORE-1475

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: .github/workflows/tag_commits_to_master.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+      with:
+        fetch-depth: '0'
+    - name: Bump version and push tag
+      uses: anothrNick/github-tag-action@1.19.0
```
Comment (pookmish) [inline .github/workflows/tag_commits_to_master.yml:14]: i'd prefer to use `K-Phoen/semver-release-action` It allows you to designate which branch to listen to merges on. this action requires `master` branch. see https://github.com/SU-SWS/react_paragraphs/blob/8.x-1.x/.github/workflows/release.yml

File: README.md [hunk 2]
```diff
@@ -25,6 +25,18 @@ Configuration
 
 Nothing special needed.
 
+Releases
+---
+
+Steps to build a new release:
+- Checkout the latest commit from the `8.x-1.x` branch.
+- Create a new branch, e.g., `release-8.1.6`.
+- Update the `composer.json` file to use the latest tagged release of all `su-sws` packages, eliminating the `@dev` dependencies.
```
Comment (pookmish) [inline README.md:34]: can we make this automatic. with github actions or something.

Comment (sherakama) [reply README.md:34]: +1 to automating the switch over on release branches but that can be a future improvement.

File: composer.json [hunk 3]
```diff
@@ -99,15 +99,15 @@
...
-        "su-sws/stanford_text_editor": "^8.1.0"
+        "su-sws/jumpstart_ui": "dev-8.x-1.x@dev",
+        "su-sws/nobots": "dev-8.x-2.x@dev",
+        "su-sws/react_paragraphs": "dev-8.x-1.x@dev",
+        "su-sws/stanford_basic": "dev-8.x-4.x@dev",
+        "su-sws/stanford_date_formats": "dev-8.x-1.x@dev",
+        "su-sws/stanford_fields": "dev-8.x-1.x@dev",
+        "su-sws/stanford_image_styles": "dev-8.x-1.x@dev",
+        "su-sws/stanford_ssp": "dev-8.x-1.x@dev",
+        "su-sws/stanford_text_editor": "dev-8.x-1.x@dev"
```
Comment (sherakama) [inline composer.json:110]: Is the `@dev` required when you are on the branch already? FWIK the `@` flag allows for an override of the minimum-stable version. Not sure if that is needed when using the branch naming convention.

Comment (imonroe) [reply composer.json:110]: When I was testing, I ran into minimum-stability problems, which the `@dev` seemed to solve.  We have minimum-stability set to `dev` above, and we also have `prefer-stable` set to `true`.  However, we are always including this profile as a dependency in another composer file, where the minimum stability may be different, and `prefer-stable` may be getting in the way.  I'll test this a little more and see if it's essential or not.

File: .github/workflows/tag_commits_to_master.yml (hunk 1 above)
Comment (imonroe) [reply .github/workflows/tag_commits_to_master.yml]: I've changed this to your preference.

File: composer.json (hunk 3 above)
Comment (imonroe) [reply composer.json]: I tested this without the `@dev` suffix, and it appears to work correctly, so I've gone ahead and removed them.

File: README.md [hunk 4]
```diff
@@ -25,6 +25,19 @@ Configuration
 
 Nothing special needed.
 
+Releases
+---
+
+Steps to build a new release:
```
Comment (pookmish) [inline README.md:31]: We should update this with the same language as the other modules we did

**Comment by pookmish** [comment]
the failing test has been a major pain. I'm hoping to get some movement on it soon. i was able to reproduce it on my local and its not a behat/circleci thing

File: README.md (hunk 4 above)
Comment (imonroe) [reply README.md:31]: Done.

**Comment by pookmish** [comment]
Do we wanna get the work in https://github.com/SU-SWS/stanford-caravan/pull/11 into this as well?

**Comment by pookmish** [comment]
Here goes nothing

**Pull request merged**